src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,51 @@
+import { useEffect, useState } from 'react';
+import { getPeople } from '../api';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import NotPeople from './NotPeople';
+import ErrorMessage from './ErrorMessage';
+import { Person } from '../types';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
   return (
     <>
       <h1 className="title">People Page</h1>
 
       <div className="block">
         <div className="columns is-desktop is-flex-direction-row-reverse">
           <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
+            {!isLoading && !isError && <PeopleFilters />}
           </div>
 
           <div className="column">
             <div className="box table-container">
-              <Loader />
-
-              <p data-cy="peopleLoadingError">Something went wrong</p>
-
-              <p data-cy="noPeopleMessage">There are no people on the server</p>
-
-              <p>There are no people matching the current search criteria</p>
-
-              <PeopleTable />
+              {isLoading ? (
+                <Loader />
+              ) : isError ? (
+                <ErrorMessage />
+              ) : !people.length ? (
+                <NotPeople />
+              ) : (
+                <PeopleTable people={people} />
+              )}
```

Comment:
The conditional rendering is a bit complex and hard to read. Consider breaking it down into smaller, well-named helper functions. This will improve the readability and maintainability of your code.